storage/engine/rocksdb.go

```diff
@@ -602,8 +602,7 @@ func (r *RocksDB) GetSSTables() SSTableInfos {
 	// We can't index into tables because it is a pointer, not a slice. The
 	// hackery below treats the pointer as an array and then constructs a slice
 	// from it.
-	const maxLen = 0x7fffffff
-	tableSlice := (*[maxLen]C.DBSSTable)(unsafe.Pointer(tables))[:n:n]
+	tableSlice := (*[(maxLen / sizeOfDBSSTable)]C.DBSSTable)(unsafe.Pointer(tables))[:n:n]
```

Comment:
The slice trickery isn't necessary here and was likely done for convenience. The alternative would be to do manually index into `tables`. Something like:

```
  tableVal := func(i int) C.DBSSTable {
    return *(*C.DBSSTable)(unsafe.Pointer(tables) + uintptr(i) * unsafe.Sizeof(C.DBSSTable))
  }
```

I doubt the above is correct, but hopefully you get the idea. This would get rid of the need for `sizeofDBSStable`.